server-common/src/main/java/org/apache/kafka/server/common/TestFeatureVersion.java

```diff
@@ -21,10 +21,10 @@
 
 public enum TestFeatureVersion implements FeatureVersion {
 
-    // TEST_1 released right before MV 3.7-IVO was released, and it has no dependencies
+    // TEST_1 released right before MV 3.7-IV0 was released, and it has no dependencies
     TEST_1(1, MetadataVersion.IBP_3_7_IV0, Collections.emptyMap()),
-    // TEST_2 released right before MV 3.8-IVO was released, and it depends on this metadata version
-    TEST_2(2, MetadataVersion.IBP_3_8_IV0, Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.IBP_3_8_IV0.featureLevel()));
+    // TEST_2 released right before MV 4.0-IV0 was released, and it depends on this metadata version
+    TEST_2(2, MetadataVersion.IBP_4_0_IV0, Collections.singletonMap(MetadataVersion.FEATURE_NAME, MetadataVersion.IBP_4_0_IV0.featureLevel()));
```

Comment:
@jolshan left a comment earlier. https://github.com/apache/kafka/pull/15673#discussion_r1630117332

Since she plans to create a new feature TEST_3 for unstable version. We probably can keep this at 3_8_IV1 since it's a stable version.

Reply:
Ah, apologies, I missed Justine's comment, I will make the change now, thank you 😊

Reply:
Thanks, Christo. Have you pushed the change?

Reply:
I put this to 3.8-IV1, but since 3.8-IV1 is now marked as a production version a test (testLatestFeaturesWithOldMetadataVersion) fails with a message like
```
test.feature.version could not be set to 2 because it depends on metadata.version level 21
```
And two tests (testUnstableTestVersion, testUnstableFeatureThrowsError) fail with
```
Expected java.lang.IllegalArgumentException to be thrown, but nothing was thrown.
```
As such, I have a preference that we keep this as IBP_4_0_IV0 which appears to exercise all scenarios successfully.

What is your opinion @jolshan? Do you want a stable version to be exercised as part of TEST_2 right now?